File: example implementations/java/convertor/src/voxel/Cube.java
package voxel;

public class Cube {
    public XYZ low; //inclusive
    public XYZ high; //inclusive
    public Colour colour = null;

    public Cube(XYZ low, XYZ high) {
        this.low = low;
        this.high = high;
    }

    public Cube(XYZ low, XYZ high, Colour c) {
        this(low, high);
        this.colour = c;
    }

    @Override
    public boolean equals(Object obj) {
        return obj instanceof Cube other && other.low.equals(low) && other.high.equals(high);
    }

    @Override
    public Cube clone() {
        return new Cube(low.clone(), high.clone(), colour);
    }
}

File: example implementations/csharp/cvox-convertor/voxel/Cube.cs
using System.Drawing;

namespace cvox_convertor.voxel
{
    public class Cube
    {
        public XYZ Low; //inclusive
        public XYZ High; //inclusive
        public Color Colour;

        public Cube(XYZ low, XYZ high) => (Low, High) = (low, high);

        public Cube(XYZ low, XYZ high, Color colour) => (Low, High, Colour) = (low, high, colour);

        public static bool operator ==(Cube cube0, Cube cube1)
        {
            return cube0.Low == cube1.Low && cube0.High == cube1.High && cube0.Colour.ToArgb() == cube1.Colour.ToArgb();
        }
        public static bool operator !=(Cube cube0, Cube cube1)
        {
            return !(cube0 == cube1);
        }
    }
}
